src/stack.ts

```diff
@@ -176,9 +176,15 @@ export function stack(m: Mark | MarkDef, encoding: Encoding<string>): StackPrope
   if (encoding[dimensionChannel]) {
     const dimensionDef = encoding[dimensionChannel];
     const dimensionField = isFieldDef(dimensionDef) ? vgField(dimensionDef, {}) : undefined;
+    const hasSameDimensionAndStackedField = dimensionField && dimensionField === stackedField;
 
-    if (dimensionField && dimensionField !== stackedField) {
+    // This change breaks cartesian tests, just pushing to start the discussion. When dev'ing locally, try removing this condition.
+    // TBD: we should only enter this branch if there's also a "color" and maybe a group channel?
+    const isPolar = isPolarPositionChannel(fieldChannel) || isPolarPositionChannel(dimensionChannel);
+
+    if (!hasSameDimensionAndStackedField && (isPolar ? dimensionChannel === fieldChannel || encoding['color'] : true)) {
       // avoid grouping by the stacked field
+      // TKTK: find out why
```

Comment:
If this works here, we may need a similar change for the code block under `dimensionOffsetChannel` .